src/os/bluestore/BlueFS.cc

```diff
@@ -901,6 +921,19 @@ int BlueFS::_read(
            << " 0x" << std::hex << off << "~" << len << std::dec
 	   << " from " << h->file->fnode << dendl;
 
+  logger->inc(l_bluefs_reads);
+  logger->inc(l_bluefs_read_bytes, len);
+  if (h->file->contents.length()) {
```

Comment:
Wondering if it's possible to have less data in file->contents than off+len? 
As far as I understand we are adding just portions of content to the cache in _flush_range so the question is if there might be a _read call for range(s) not handled by _flush_range yet.